llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7456,6 +7456,9 @@ LoopVectorizationPlanner::executePlan(
   if (!IsEpilogueVectorization)
     VPlanTransforms::optimizeForVFAndUF(BestVPlan, BestVF, BestUF, PSE);
 
+  BestVPlan.setName("Final VPlan");
```

Comment:
Yep, will move printing "Executing best plan..." down!

> Regarding VPlan's name - "Initial" VPlans are currently dumped following buildVPlansWithVPRecipes() which also optimize()'s them, except for the last optimizeForVFAndUF(). Perhaps the processing stage a VPlan is currently in should be better maintained than in a name string.

Perhaps Initial = after initial construction, Optimized = during most VPlan transforms, Final = just before execution?

Reply:
> Perhaps Initial = after initial construction, Optimized = during most VPlan transforms, Final = just before execution?

SGTM. Although the name may be used more persistently to associate a VPlan with the fixed original loop, complemented with optimization decisions (e.g., VF, UF), rather than keep track of which optimizations were processed.